aten/src/ATen/native/TensorShape.cpp

```diff
@@ -513,6 +513,40 @@ std::vector<Tensor> chunk(const Tensor& self, int64_t chunks, int64_t dim) {
   }
 }
 
+static void check_tensor_split_dim(const Tensor& self, int64_t dim) {
+  TORCH_CHECK(self.dim() > 0, "expected at least a 1-dimensional tensor");
+  TORCH_CHECK(dim < self.dim(), "expected dim to be less than self.dim()");
+}
+
+std::vector<Tensor> tensor_split(const Tensor& self, int64_t sections, int64_t dim) {
+  check_tensor_split_dim(self, dim);
+  TORCH_CHECK(sections > 0, "number of sections must be larger than 0, got ", sections);
+  std::vector<Tensor> splits(sections);
+  int64_t min_split_size = self.size(dim) / sections;
```

Comment:
Should this dim be wrapped before to use to support values like dim=-1?

Reply:
Good idea. The `maybe_wrap_dim` call will perform error checking, so I can remove the `dim < self.dim()` check from above.